### ISSUE TITLE
Table formatting for schedules

### ISSUE THREAD

**Issue by emiliom** [oceanhackweek/oceanhackweek.github.io#169]
@abkfenris do you have a suggestion for the table template to use for the schedules? What we did last year and the year before looked reasonable, but it's based on HTML tables so it's [not for the faint of heart](https//github.com/oceanhackweek/ohw-resources/blob/master/docs/schedule.md). A plain markdown table doesn't give us enough control on column widths and such, and ends up looking ugly and un-user-friendly.

Here's what the HTML table looks like from last year: https://oceanhackweek.github.io/ohw-resources/schedule/#main-virtual-event

And here's a snippet of the HTML table code used:

```html
<table class ="defTable">
  <thead>
    <tr>
      <th>Time</th>
      <th>Event</th>
      <th>Leads</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
      8:00AM - 9:00AM
      <br>PDT
      </td>
      <td>
        <strong>Check-in, ask us anything</strong>
        <br>Start times in other timezones:
        <br>11:00am EDT / 15:00 UTC / 1:00am AEST (+1 day)
      </td>
      <td>
        <ul>
          <li>Matthew Biddle</li>
          <li>Alex Kerney</li>
          <li>Emilio Mayorga</li>
        </ul>
      </td>
    </tr>
  </tbody>
</table>
```

**Comment by abkfenris** [comment]
I'm thinking Google Sheets + iframes. Let me test that out.

**Comment by abkfenris** [comment]
Really quick and dirty attempt from this doc: https://docs.google.com/spreadsheets/d/1ncIrjRGjw7o4tiy2LrdE8mZohykThxi_kaow8v-g4lg/edit#gid=645587196 <img width="1135" alt="image" src="https://user-images.githubusercontent.com/1296209/181400234-f697ee4e-be0a-4057-9a6f-311c2a5ef6de.png">

It would be nice to have both a compact view of the schedule (see all days at once), and a full expanded one for each satellite/global. If we start by producing both of those schedules for the global, then each satellite can duplicate the tab and then make modifications as necessary without needing to dig into the HTML to modify their pages. On the schedules page we have the global info, and links to each satellite page.

**Comment by emiliom** [comment]
Wow, that's thinking outside the box (or inside the iframe?)! That could definitely work, as you described. I imagine it'll be slower to load the page, and I wonder how it looks on a phone (easy to test, though). How quickly does it load?

I've done some digging for alternatives mid-way between simple, limited markdown tables and full blown (but nasty) HTML. The best option I found is `rST` [flat-tables](https://return42.github.io/linuxdoc/linuxdoc-howto/table-markup.html#flat-table) from the linuxdoc Sphinx extension. Here is one example, from this [nice blog post](https://koen.vervloesem.eu/blog/using-flat-tables-in-restructuredtext-with-sphinx-for-column-and-row-spans/):

```rst
.. flat-table:: Characteristics of the BLE badge
   :header-rows: 1

   * - Service
     - Characteristic
     - Properties
   * - :rspan:`2` 0xfee7
     - 0xfec7
     - WRITE
   * - 0xfec8
     - INDICATE
   * - 0xfec9
     - READ
   * - 0xfee0
     - 0xfee1
     - NOTIFY, READ, WRITE
```

The rendered output:
![rendered flat-tables](https://koen.vervloesem.eu/images/rst-flat-table-example.png)

It looks like it can be [used from MyST directly](https://gdevops.gitlab.io/tuto_documentation/formats/input/rest_sphinx/tables/flat_tables/flat_tables.html), once the extension is installed. Otherwise, it can be wrapped in a `{eval-rst}` code block.

- **Pros:** inline text, so it'll be very lightweight; relatively easy to learn and use; less overhead
- **Cons:** still way less transparent or user friendly to people entering and maintaining the table info (though I guess it's not much more complex than markdown in general); a table can't be reused and displayed in multiple pages -- unless it's maintained as a separate rST file that's loaded/included; the Google Sheet can be used as a stand-alone document, and this one can't

A third alternative (b/c it's never enough ...) is to use an external system like https://sessionize.com (or https://sched.com, but it has no free plan). sessionize allows for [embedding of the schedule](https://sessionize.com/developers) into your own web page, and you can pick different schedule views. I've never actually used it myself, but I'd guess data entry is user friendly.

What do you think? I'll go with whatever you recommend.

**Comment by abkfenris** [comment]
Ehh, either markdown tables or rST flat tables are a whole new syntax to learn, so between those and HTML I would stay with HTML as it still is going to be us editing the tables in that case.

However, if we go with Google Sheets, then we don't have to be the ones editing the tables every time a satellite makes a tweak. Instead they can do that themselves, and it's immediately updated. It didn't seem to slow down loading notably. 

I'll see if I can draft up a PR with some Google sheets schedules this morning, but I only have a few hours before I need to head out for the weekend.

**Comment by abkfenris** [comment]
I didn't get a chance to draft anything before running out the door.

**Comment by emiliom** [comment]
Thanks. Google Sheets it is, then. 

I assume that if there is a link in a cell, it will be clickable so that a user can follow the link? That's to replicate that convenience we had previously:
![image](https://user-images.githubusercontent.com/742403/181853363-e1cc632d-1193-48b7-8ce5-d3a23e162670.png)
Based on a test I did on a Google Sheet, it looks like a cell will only expose one link for clicking, even if it contains two or more. We can work with that.

**Comment by abkfenris** [comment]
Ok, draft with the compact schedule table and expanded headers for more detail at #172

**Comment by abkfenris** [comment]
Closed by #172
